lib/asciidoctor/chart/docinfo_processor.rb

```diff
@@ -42,9 +35,55 @@ class DocinfoProcessor < ::Asciidoctor::Extensions::DocinfoProcessor
         </style>
       HTML
 
-      def process doc
+      def process(doc)
         engines = doc.x_chart[:engines]
-        (engines.map {|engine| ASSETS[engine.to_sym] }.to_a + [DEFAULT_STYLE]).join("\n")
+        result = []
+        engines.each do |engine|
+          # noinspection RubyCaseWithoutElseBlockInspection
+          case engine
+          when 'c3js'
+            result << c3js(doc)
+          when 'chartjs'
+            result << chartjs(doc)
+          when 'chartist'
+            result << chartist(doc)
+          end
+        end
```

Comment:
since we now have methods with the same name as the engine, we could use `send` to call the method.

**Pseudo-code**

```rb
(engines.map {|engine| send(engine, doc) }.to_a + [DEFAULT_STYLE]).join("\n")
```